plone/app/contenttypes/migration/field_migrators.py

```diff
@@ -17,17 +17,24 @@ def migrate_simplefield(src_obj, dst_obj, src_fieldname, dst_fieldname):
     to the target-object. The only transform is a safe_unicode of the value.
     """
     field = src_obj.getField(src_fieldname)
-    if field:
-        at_value = field.get(src_obj)
+    try:
+        if field:
+            # Check for existence first
+            field.getStorage(src_obj).get(field.getName(), src_obj)
+            at_value = field.get(src_obj)
+        else:
+            at_value = getattr(src_obj, src_fieldname)
+            if at_value and hasattr(at_value, '__call__'):
+                at_value = at_value()
+    except AttributeError:
+        # The value is "unset" on the old instance and so should be unset on
+        # the new instance
+        pass
```

Comment:
Would it not be easier to just return here?

Reply:
I thought about that too, I have no strong preference.

Reply:
I would just return and remove the else.

Reply:
I find it more readable, the blocks indicate which part of the "condition" the code is in.  But I don't feel strong enough about it to object if you want to change it.

Reply:
No, I am fine!